### ISSUE TITLE
Is props destructuring never allowed?

### ISSUE THREAD

**Issue by elsassph** [solidjs-community/eslint-plugin-solid#191]
I'd think this should be allowed but we get a reactivity error inside `createMemo` and `createEffect`:
```typescript
  const item = createMemo(() => {
    const { item } = props; // <-- solid/reactivity error
    return item;
  });
```

I guess this is a trivial case as it means maybe supporting more complex cases like
```typescript
  const getItem = () => {
    const { item } = props; // <-- solid/reactivity error
    return item;
  }
  const item = createMemo(getItem);

  // or
  <Child item={getItem()} />
```

**Comment by ryansolid** [comment]
Confirmed on current main, and agreed it's a false positive as reported: `const { item } = props` *inside* `createMemo`/`createEffect` re-runs with the tracked scope, so the destructured value can't go stale within it. Destructuring outside tracked scopes remains correctly flagged. Queued for a 0.16.1 look alongside the other reactivity precision fixes.

**Comment by ryansolid** [comment]
Fixed in v0.16.1. Destructuring props inside a tracked scope (`createMemo`, `createEffect`, event handlers) no longer warns — the destructuring re-runs on updates there. Destructuring at component setup level still warns, since that genuinely snapshots values.
